ExRam.Gremlinq.Core/Models/Metadata/IMetadataStore.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Reflection;
+
+namespace ExRam.Gremlinq.Core
+{
+    public interface IMetadataStore : ICloneable
+    {
+        PropertyMetadata TryGetPropertyMetadata(Type elementType, PropertyInfo property);
+
+        void UpdatePropertyMetadata(Type type, PropertyInfo propertyInfo, Action<PropertyMetadata> updateAction);
```

Comment:
No mutablility please.

Reply:
I updated this Update logic into the ElementBuilder where it should only be updated during the 'Configure' method. The Update method is actually now hidden from the end users.